java-frontend/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -175,4 +194,22 @@ private static boolean isAnnotatedCheckForNull(MethodInvocationTree syntaxNode)
     return syntaxNode.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull");
   }
 
+  @Override
+  public void checkEndOfExecution(CheckerContext context) {
+    reportIssues();
+  }
+
+  @Override
+  public void interruptedExecution(CheckerContext context) {
+    reportIssues();
+  }
+
+  private void reportIssues() {
+    Iterator<NPEContext> i = detectedIssues.iterator();
+    while (i.hasNext()) {
+      NPEContext context = i.next();
+      reportIssue(context.symbolicValue, context.tree, context.node);
+      i.remove();
```

Comment:
I rewrote the loop and I clear collection afterwards. The reason for clearing is to avoid holding a reference to something which can already be garbage collected, given the live scope of check instance is bigger that the one of the exploded graph.